### ISSUE TITLE
Data-stream configuration for elasticsearch output causes fluentd to crashloop

### ISSUE THREAD

**Issue by Synehan** [banzaicloud/logging-operator#878]
> Bugs should be filed for issues encountered whilst operating logging-operator.
> You should first attempt to resolve your issues through the community support
> channels, e.g. Slack, in order to rule out individual configuration errors. [#logging-operator](https://slack.banzaicloud.io/)
> Please provide as much detail as possible.

**Describe the bug**:
fluent-plugin-elasticsearch is not working any if using data-stream. this plugin have a bug in the version v5.1.1 (https://github.com/uken/fluent-plugin-elasticsearch/issues/925) and was corrected in v5.1.2. 

When using data-stream fluend crashloop with the following message:

```bash
/usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:169:in `block in valid_characters?': undefined method `include?' for :data_stream_name:Symbol (NoMethodError)
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:169:in `each'
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:169:in `each'
     from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:169:in `any?'
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:169:in `valid_characters?'
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:158:in `valid_data_stream_parameters?'
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:55:in `block in validate_data_stream_parameters'
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:54:in `each'
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:54:in `validate_data_stream_parameters'
    from /usr/lib/ruby/gems/2.7.0/gems/fluent-plugin-elasticsearch-5.1.1/lib/fluent/plugin/out_elasticsearch_data_stream.rb:32:in `configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/plugin.rb:178:in `configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/agent.rb:132:in `add_match'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/agent.rb:74:in `block in configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/agent.rb:64:in `each'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/agent.rb:64:in `configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/label.rb:31:in `configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/root_agent.rb:143:in `block in configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/root_agent.rb:143:in `each'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/root_agent.rb:143:in `configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/engine.rb:105:in `configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/engine.rb:80:in `run_configure'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/supervisor.rb:663:in `run_supervisor'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/lib/fluent/command/fluentd.rb:344:in `<top (required)>'
    from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require'
    from /usr/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require'
    from /usr/lib/ruby/gems/2.7.0/gems/fluentd-1.13.3/bin/fluentd:15:in `<top (required)>'
```

**Expected behaviour**:
Logs start shipping to ES

**Steps to reproduce the bug**:
Create an ouptut with es plug in and the following parameters

```yaml
    data_stream_enable: true
    data_stream_name: "e7-stream"
````

**Environment details**:
- Kubernetes version (e.g. v1.15.2): v1.21.2
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): AKS
- logging-operator version (e.g. 2.1.1): 3.15.0
- Install method (e.g. helm or static manifests): helm
- Logs from the misbehaving component (and any other relevant logs):
- Resource definition (possibly in YAML format) that caused the issue, without sensitive data:

/kind bug

**Comment by ahma** [comment]
This package is updated in [3.16.0](https://github.com/banzaicloud/logging-operator/releases/tag/3.16.0 )  release. If this isn't solve, please reopen this issue !

**Comment by Synehan** [comment]
Totally solved with 3.16.0 thanks !
